lib/ansible/modules/cloud/vmware/vmware_host_acceptance.py

```diff
@@ -0,0 +1,206 @@
+            if cluster_obj:
+                self.hosts = [host for host in cluster_obj.host]
+            else:
+                module.fail_json(changed=False, msg="Cluster '%s' not found" % cluster_name)
+        elif esxi_host_name:
+            esxi_host_obj = self.find_hostsystem_by_name(host_name=esxi_host_name)
+            if esxi_host_obj:
+                self.hosts = [esxi_host_obj]
+            else:
+                module.fail_json(changed=False, msg="ESXi '%s' not found" % esxi_host_name)
+        self.desired_state = self.params.get('state', 'list')
+        self.hosts_facts = {}
+        self.acceptance_level = self.params.get('acceptance_level')
+
+    def gather_acceptance_facts(self):
+        for host in self.hosts:
+            self.hosts_facts[host.name] = dict(level='', error='NA')
+            host_image_config_mgr = host.configManager.imageConfigManager
+            if host_image_config_mgr:
+                try:
+                    self.hosts_facts[host.name]['level'] = host_image_config_mgr.HostImageConfigGetAcceptance()
+                except vim.fault.HostConfigFault as e:
+                    self.hosts_facts[host.name]['error'] = to_native(e.msg)
+
+    def set_acceptance_level(self):
+        change = []
+        for host in self.hosts:
+            host_changed = False
+            if self.acceptance_level != self.hosts_facts[host.name]['level']:
+                host_image_config_mgr = host.configManager.imageConfigManager
+                if host_image_config_mgr:
+                    try:
+                        host_image_config_mgr.UpdateHostImageAcceptanceLevel(newAcceptanceLevel=self.acceptance_level)
+                        host_changed = True
+                        self.hosts_facts[host.name]['level'] = host_image_config_mgr.HostImageConfigGetAcceptance()
+                    except vim.fault.HostConfigFault as e:
+                        self.hosts_facts[host.name]['error'] = to_native(e.msg)
+
+            change.append(host_changed)
+        self.module.exit_json(changed=any(change), facts=self.hosts_facts)
+
+    def reset_acceptance_level(self):
+        change = []
+        for host in self.hosts:
+            host_changed = False
+            if self.hosts_facts[host.name]['level'] != 'partner':
+                host_image_config_mgr = host.configManager.imageConfigManager
+                if host_image_config_mgr:
+                    try:
+                        host_image_config_mgr.UpdateHostImageAcceptanceLevel(newAcceptanceLevel='partner')
+                        host_changed = True
+                        self.hosts_facts[host.name]['level'] = host_image_config_mgr.HostImageConfigGetAcceptance()
+                    except vim.fault.HostConfigFault as e:
+                        self.hosts_facts[host.name]['error'] = to_native(e.msg)
+            change.append(host_changed)
+        self.module.exit_json(changed=any(change), facts=self.hosts_facts)
```

Comment:
remove reset_acceptance_level ; modify and re-use set_acceptance_level instead (reset_acceptance_level and set_acceptance_level have just 1 minor difference).

Reply:
Done.